Simplenote/TagListState.swift

```diff
@@ -108,7 +108,7 @@ extension TagListState {
     ///
     @objc
     func tag(atIndex index: Int) -> Tag? {
-        guard case let .tag(tag) = rowAtIndex(index) else {
+        guard let row = rowAtIndex(index), case let .tag(tag) = row else {
```

Comment:
The compiler required `rowAtIndex(_:)` `Optional` return value to be unwrapped, although I don't understand why this failure appeared now, since it's not related with Tracks 🤔

Reply:
Absolutely a mystery to me as well, this change is unrelated to anything else. I suspect it's perhaps a glitch in the compiler (since one of the changes is the supported archs right?).

No big deal, it's a valid change!! thank you Gio!!

Reply:
I think that's a change in a recent Swift compiler update?
And another possibility to fix it would have been:
```swift
        guard case let .tag(tag)? = rowAtIndex(index) else {
```

Or, as I usually prefer to write it (but to each their own):
```swift
        guard case .tag(let tag)? = rowAtIndex(index) else {
```

Which both are the shorthand form for `guard case Optional.some(TagListRow.tag(let tag)) = rowAtIndex(index)`, which is a correct pattern matching since `rowAtIndex` indeed returns an `Optional<TagListRow>` and the types of both operands of the `case lhs = rhs` has to match.